android/app/src/main/kotlin/coop/polypoly/polypod/polyOut/PolyOut.kt

```diff
@@ -2,24 +2,30 @@ package coop.polypoly.polypod.polyOut
 
 import android.content.Context
 import coop.polypoly.polypod.Preferences
+import coop.polypoly.polypod.polyNav.ZipTools
 import java.io.File
 import java.nio.ByteBuffer
-import java.util.zip.ZipFile
 
 open class PolyOut(
     val context: Context
 ) {
     private val fsPrefix = "polypod://"
+
     open suspend fun readFile(path: String): ByteArray {
         if (path == "") {
             throw Error("Empty path in PolyOut.readFile")
         }
         val fs = Preferences.getFileSystem(context)
-        val entryPathStart = path.indexOf('/', fsPrefix.length)
-        val zipId = path.substring(0, entryPathStart)
-        val entryPath = path.substring(entryPathStart + 1)
-        val zip = ZipFile(File(fs[zipId]))
-        return zip.getInputStream(zip.getEntry(entryPath)).readBytes()
+
+        val filePath = context.filesDir.absolutePath + "/" + path.replace(
+            "://", ":/"
```

Comment:
This part freaks me out a little bit, for three reasons:

1. It's duplicated below :D
2. I don't quite get it :D - How I understand everything, we actually store files from an imported archive into a directory called `polypod://$UUID` - is that about correct? Then why do we need to change `polypod://$UUID/$path` to `polypod:/$UUID/$path` here?
3.  A directory name with slashes (or even a colon) seems a bit like asking for trouble - couldn't we have something like `imported/$UUID` or something? Then we could just remove the entire `fsPrefix` from the beginning of the path, right?

Reply:
I keep the prefix for compatibility with iOS. But yes, this does look weird. Let me redesign a bit...

Reply:
done.